components/renku_data_services/session/models.py

```diff
@@ -25,6 +25,44 @@ class EnvironmentKind(StrEnum):
     CUSTOM = "CUSTOM"
 
 
+class EnvironmentImageSource(StrEnum):
+    """The source of the environment image."""
+
+    image = "image"
+    build = "build"
+
+
+class BuilderVariant(StrEnum):
+    """The type of environment builder."""
+
+    conda = "conda"
+    pip = "pip"
```

Comment:
The builder will just build anything Python, so the choice is only "python" at the moment.

```suggestion
    python = "python"
```